mapper/thin_edge_json/src/serialize.rs

```diff
@@ -0,0 +1,112 @@
+use chrono::offset::FixedOffset;
+use chrono::DateTime;
+use std::io::Write;
+
+use crate::measurement::GroupedMeasurementVisitor;
+pub struct ThinEdgeJsonSerializer {
+    buffer: Vec<u8>,
+    is_within_group: bool,
+    needs_separator: bool,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ThinEdgeJsonSerializationError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error(transparent)]
+    MeasurementCollectorError(#[from] MeasurementStreamError),
+}
+#[derive(thiserror::Error, Debug)]
+pub enum MeasurementStreamError {
+    #[error("Unexpected time stamp within a group")]
+    UnexpectedTimestamp,
+
+    #[error("Unexpected end of data")]
+    UnexpectedEndOfData,
+
+    #[error("Unexpected end of group")]
+    UnexpectedEndOfGroup,
+
+    #[error("Unexpected start of group")]
+    UnexpectedStartOfGroup,
+}
+
+impl ThinEdgeJsonSerializer {
+    pub fn new() -> Self {
+        ThinEdgeJsonSerializer {
+            buffer: Vec::new(),
+            is_within_group: false,
+            needs_separator: false,
+        }
+    }
+    pub fn start(&mut self) -> Result<(), ThinEdgeJsonSerializationError> {
+        self.buffer.push(b'{');
+        self.needs_separator = false;
+        Ok(())
+    }
+
+    pub fn end(&mut self) -> Result<(), ThinEdgeJsonSerializationError> {
+        if self.is_within_group {
+            return Err(MeasurementStreamError::UnexpectedEndOfData.into());
+        }
+
+        self.buffer.push(b'}');
+        Ok(())
+    }
+
+    pub fn get_searialized_tej(&mut self) -> Vec<u8> {
+        self.buffer.clone()
+    }
```

Comment:
Many comments here.

* I don't like the name (beyond the typo). Why a new abbreviation? Why not simply `bytes()` ?
* The `self` parameter should not be borrowed but owned.
    * We are trying to reduce memory pressure. Hence, it's better to avoid an unnecessary buffer  `clone()`.
    * It make not sense to call this method twice.
 * The `end()` method needs to be called.
    
Hence I would write:

```
pub fn bytes(mut self) -> Result<Vec<u8>, ThinEdgeJsonSerializationError> {
        self.end()?;
        self.buffer
    }
```